clis/tcli/src/bin/tclid/main.rs

```diff
@@ -0,0 +1,189 @@
+#![deny(missing_docs)]
+// TCLID doc rs documentation introduction:
+#![doc = include_str!["./doc/tclid_documentation.md"]]
+
+//! Main and code for the API which is used for sending commands to libtelio which runs on a daemon.
+
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result};
+use clap::Parser;
+use tokio::time::Duration;
+
+mod coms;
+mod daemon;
+mod tclid;
+mod telio_cli_interface;
+
+use coms::DaemonIPC;
+use daemon::{Daemon, ProcessType};
+use telio_cli_interface::TelioCliInterface;
+
+/// Name for the local socket file which will be used for inter process communication.
+const DAEMON_IPC_SOCKET: &str = "tclid.sock";
+/// Path where all the logs, local socket and pid file will be stored, relative to the path of this binary.
+const TCLID_WD_PATH: &str = "/var/run/tclid_wd";
+
+// TODO Add meshnet examples.
+// TODO Add --restart flag which would restart the whole daemon before executing the provided command for faster iterating when debugging.
+// TODO figure out logging, because multiple tracing subscribers can't be used. Maybe reroute STDIO of daemon?
+// TODO Maybe add auto completion support? (probably won't work due to mixed parsing implementation)
+
+/// Argument structure for the TCLID API used by Clap to parse the arguments.
+#[derive(Parser)]
+#[clap(disable_help_flag = true)]
+struct Args {
+    /// This flag is set when --help or -h is passed, but help is disabled for this struct,
+    /// because we want to retrieve the help message from the original TCLI parser to keep things simple.
+    /// So help is handled separately according to this flag.
+    #[clap(short = 'h', long = "help")]
+    help: bool,
+    /// Used to pass in features to the TCLID daemon. Note that the TCLID daemon only accepts
+    /// feature flags on startup.
+    #[clap(short, long)]
+    features: Option<String>,
+    /// Holds all the other arguments which will be forwarded to the TCLI parser.
+    input: Vec<String>,
+}
+
+/// Main of the whole TCLID package.
+fn main() -> Result<()> {
+    let args: Args = Args::parse();
+
+    match run_api(args) {
+        Ok(()) => Ok(()),
+        Err(e) => {
+            #[cfg(unix)]
+            // For convenience printing stderr of the daemon to inform the user that something's wrong.
+            if let Ok(stderr) = Daemon::get_stderr() {
+                if !stderr.is_empty() {
+                    println!("Daemon stderr: {}", stderr);
```

Comment:
Maybe it would be nice to use some sort of logging, e.g. even https://docs.rs/simple_logger/latest/simple_logger/ instead of `println!`?  :thinking:

Reply:
I've moved the tracing subscriber from Cli to TCLI main and TCLID daemon, so now println is only used to respond to user commands